app/src/ui/discard-changes/discard-changes-dialog.tsx

```diff
@@ -65,15 +66,17 @@ export class DiscardChanges extends React.Component<
           <p>
             Changes can be restored by retrieving them from the {trashName}.
           </p>
-          <Checkbox
-            label="Do not show this message again"
-            value={
-              this.state.confirmDiscardChanges
-                ? CheckboxValue.Off
-                : CheckboxValue.On
-            }
-            onChange={this.onCheckboxChanged}
-          />
+          {!this.props.forceConfirmation ? (
```

Comment:
The `!` here made me question what the right flow was. I gather we're aiming for:

 - if `forceConfirmation` is true, don't show the checkbox
 - otherwise, show the checkbox with the confirmation

If that's the case, could we make this a bit easier to follow - extract it to a function rather than inlining?